lib/src/main/java/org/asamk/signal/manager/Manager.java

```diff
@@ -189,6 +191,9 @@
     private final SignalServiceMessageReceiver messageReceiver;
     private final ClientZkProfileOperations clientZkProfileOperations;
 
+    private final ExecutorService executor = new ThreadPoolExecutor(0, 10, 10, TimeUnit.MILLISECONDS, new LinkedBlockingQueue<Runnable>());
```

Comment:
Initializing the ThreadPoolExecutor with a LinkedBlockingQueue effectively makes this a single thread executor.
See javadoc:
>  Unbounded queues. Using an unbounded queue (for example a LinkedBlockingQueue without a predefined capacity) will cause new tasks to wait in the queue when all corePoolSize threads are busy. Thus, no more than corePoolSize threads will ever be created. (And the value of the maximumPoolSize therefore doesn't have any effect.) This may be appropriate when each task is completely independent of others, so tasks cannot affect each others execution; for example, in a web page server. While this style of queuing can be useful in smoothing out transient bursts of requests, it admits the possibility of unbounded work queue growth when commands continue to arrive on average faster than they can be processed.

I'm fine with that, but then you can just use `Executors.newSingleThreadExecutor()` :)
Also using a coreSize of 1 is probably a good idea (which newSingleThreadExecutor already does), the thread will be started lazily anyway.

Reply:
Ups, I've trusted the answer of StackOverflow and just skimmed the Javadoc afterward - classic mistake ^^'

I think a single thread might be a bottleneck in scenarios where one wants to send multiple messages on a mediocre internet connection. Having read the Javadoc around ThreadPoolExecutor and Executors more carefully, I suggest using a `Executors.newCachedThreadPool()`. Since the number of threads is virtually unbounded, the behavior is almost similar to before the change - with the benefit that threads are reused and that the thread pool is shutdown on closing.